homeassistant/components/synology_dsm/camera.py

```diff
@@ -30,39 +36,73 @@ async def async_setup_entry(
     if SynoSurveillanceStation.CAMERA_API_KEY not in api.dsm.apis:
         return
 
-    surveillance_station = api.surveillance_station
-    await hass.async_add_executor_job(surveillance_station.update)
-    cameras = surveillance_station.get_all_cameras()
-    entities = [SynoDSMCamera(api, camera) for camera in cameras]
-
-    async_add_entities(entities)
-
-
-class SynoDSMCamera(SynologyDSMEntity, Camera):
+    async def async_update_data():
+        """Fetch all camera data from api."""
+        surveillance_station = api.surveillance_station
+        try:
+            async with async_timeout.timeout(10):
+                await hass.async_add_executor_job(surveillance_station.update)
+        except SynologyDSMAPIErrorException as err:
+            raise UpdateFailed(f"Error communicating with API: {err}") from err
+
+        data = {}
+        for camera in surveillance_station.get_all_cameras():
+            data[camera.id] = camera
+        return data
+
+    coordinator = DataUpdateCoordinator(
```

Comment:
Setup the coordinator in `__init__.py` and store it in hass.data[DOMAIN][config_entry_id]....

Then use it for the camera switch as well since it also uses the `self._api.surveillance_station` data

Reply:
yeah for sure, you are right ... I have moved the coordinator into `__init__.py` and extended about data gathering for `SynoDSMSurveillanceHomeModeToggle`